x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/wrapper.tsx

```diff
@@ -45,9 +50,17 @@ interface AiForSOCAlertsTabProps {
  * It renders a loading skeleton while packages are being fetched and while the dataView is being created.
  */
 export const AiForSOCAlertsTab = memo(({ id, query }: AiForSOCAlertsTabProps) => {
-  const { data } = useKibana().services;
-  const [dataView, setDataView] = useState<DataView | undefined>(undefined);
-  const [dataViewLoading, setDataViewLoading] = useState<boolean>(true);
+  const { dataView: oldDataView, loading: oldDataViewLoading } = useCreateDataView({
+    dataViewSpec: AI4SOC_DATA_VIEW_SPEC,
+  });
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+  // TODO: use alert only data view when it is ready
+  // https://github.com/elastic/security-team/issues/12589
+  const { dataView: experimentalDataView, status } = useDataView(SourcererScopeName.detections);
+
+  const loading = newDataViewPickerEnabled ? status !== 'ready' : oldDataViewLoading;
```

Comment:
Can we keep this variable named `dataViewLoading` for clarity? Then you wouldn't have to change the line a bit below in the `EuiSkeletonRectangle`